dht/dht.go

```diff
@@ -288,32 +321,52 @@ func (dht *DHT) iterativeStore(value string) (hash store.Key, err error) {
 	}
 
 	if len(stored) > 0 {
-		logStoredAt(hash, stored)
+		logStoredAt(hash, stored...)
 	}
 
 	return
 }
 
 func (dht *DHT) iterativeFindValue(hash store.Key) (value string, err error) {
 	call := NewFindValueCall(hash)
-	if _, err = dht.walk(call); err == nil {
+	closest, err := dht.walk(call)
+	if err != nil {
+		return
+	}
+
+	if call.value != "" {
 		value = call.value
+	} else {
+		err = fmt.Errorf("Couldn't find any value with the hash: %v", hash)
+		return
 	}
+
+	// Store at the closest node that did not return any value.
```

Comment:
Is this part of Kademlia? 🙄 😮

Reply:
Yep
![2019-09-29-195202_845x56_scrot](https://user-images.githubusercontent.com/4541181/65836838-9ddd4d80-e2f2-11e9-8f2c-2aafb25ab729.png)